### PR TITLE
python3Packages.cirq-core: 0.11.0 -> 0.12.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.12.0

Change log: https://github.com/quantumlib/Cirq/releases/tag/v0.12.0

Upstream restructured some parts and this requires additional packages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 138502` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>python38Packages.cirq</li>
    <li>python38Packages.cirq-aqt</li>
    <li>python38Packages.cirq-core</li>
    <li>python38Packages.cirq-google</li>
    <li>python38Packages.cirq-ionq</li>
    <li>python38Packages.cirq-pasqal</li>
    <li>python38Packages.cirq-rigetti</li>
    <li>python38Packages.cirq-web</li>
    <li>python38Packages.pyquil</li>
    <li>python38Packages.qcs-api-client</li>
    <li>python38Packages.rfc3339</li>
    <li>python38Packages.rpcq</li>
    <li>python39Packages.cirq</li>
    <li>python39Packages.cirq-aqt</li>
    <li>python39Packages.cirq-core</li>
    <li>python39Packages.cirq-google</li>
    <li>python39Packages.cirq-ionq</li>
    <li>python39Packages.cirq-pasqal</li>
    <li>python39Packages.cirq-rigetti</li>
    <li>python39Packages.cirq-web</li>
    <li>python39Packages.pyquil</li>
    <li>python39Packages.qcs-api-client</li>
    <li>python39Packages.rfc3339</li>
    <li>python39Packages.rpcq</li>
  </ul>
</details>

**Comment by risicle** [comment]
`python39Packages.rpcq` on macos 10.15:

```
=========================== short test summary info ============================
FAILED rpcq/test/test_auth.py::test_client_backlog - TimeoutError: Timeout on...
FAILED rpcq/test/test_rpc.py::test_client_backlog - TimeoutError: Timeout on ...
=================== 2 failed, 32 passed in 117.45s (0:01:57) ===================
```

This appears to be a time-sensitive test anyway, which are often good to disable due to their flakiness.

**Comment by drewrisinger** [comment]
Thanks for doing this, I know it's a lot of work. FWIW, these packages also exist in [my NUR](https://github.com/drewrisinger/nur-packages) if you need to reference them/check for disabled tests, I usually update that quickly b/c there's less review, haven't had time to backport these to mainstream nixpkgs. I'll try to get to reviewing this a little later.

File: pkgs/development/python-modules/cirq-web/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  propagatedBuildInputs = [
+    cirq-core
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  # cirq's importlib hook doesn't work here
+  #pythonImportsCheck = [ "cirq_web" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cirq-web/default.nix:24]: Why?

**Pull request merged**